docs/source/troubleshooting.rst

```diff
@@ -65,22 +65,165 @@ try deleting or renaming that kernelspec folder to expose the default.
 If your problem is with another kernel, not the Python one we maintain,
 you may need to look for support about that kernel.
 
+Python Environments
+-------------------
+Multiple python environments, whether based on Anaconda or Python Virtual environments,
+are often the source of reported issues.  In many cases, these issues stem from the
+Notebook server running in one environment, while the kernel and/or its resources,
+derive from another environment.  Indicators of this scenario include:
+
+* ``import`` statements within code cells producing ``ModuleNotFound`` exceptions.
+* General kernel startup failures exhibited by nothing happening when attempting
+  to execute a cell.
+
+In these situations, take a close look at your environment structure and ensure all
+packages required by your notebook's code are installed in the correct environment.
+If you need to run the kernel from different conda environments than your Notebook
```

Comment:
I think we should reference [these docs](https://ipython.readthedocs.io/en/stable/install/kernel_install.html#kernels-for-different-environments) **before** `nb_conda_kernels`. It describes the Jupyter supported way for installing kernels for other conda environments.

Reply:
Thanks Zach - I was not aware of that.  An update has been added.